runtime/smithy-client/common/src/aws/smithy/kotlin/runtime/client/config/RetryMode.kt

```diff
@@ -18,12 +18,14 @@ public enum class RetryMode {
     LEGACY,
 
     /**
-     * The standard retry mode. With this, the client will use the [StandardRetryStrategy][aws.smithy.kotlin.runtime.retries.StandardRetryStrategy]
+     * The standard retry mode. With this, the client will use the
+     * [StandardRetryStrategy][aws.smithy.kotlin.runtime.retries.StandardRetryStrategy].
      */
     STANDARD,
 
     /**
-     * Not implemented yet. https://github.com/awslabs/aws-sdk-kotlin/issues/701
+     * Adaptive retry mode. With this, the client will use the
```

Comment:
**Question**: When would we choose one over the other? This seems like guidance in the developer guide would be good. In fact I haven't seen much from other SDKs here either on guidance and tuning. We should do better in our own guide.